### ISSUE TITLE
Allow only selected hours in calendar

### ISSUE THREAD

**Issue by roman5527** [fomantic/Fomantic-UI#1351]
Hi, can i set in calendar example only hours from 7:00 to 16:00 ? and other hours in day are disabled ? 
can i set it ? 

tahnks

**Comment by lubber-de** [comment]
Disabling individual hour or minute cells are currently not implemted.
However, you can fetch the selected hour before changing the calendar field to prevent the selection

```javascript
$('.ui.calendar').calendar({
 type: 'time',
 onBeforeChange: function (date, text, mode) {
    if (date.getHours() <7 || date.getHours() >15) {
      $('body').toast({
        message: 'Only times between 7:00 and 16:00 are allowed'
      });
      return false; // returning false prevents the selection
    }
  }
});
```

**Comment by roman5527** [comment]
Hi, thanks . And can you add this features ? 
thanks

**Comment by lubber-de** [comment]
Sure, somebody just needs to donate some spare time to code this an provide a PR :wink:

**Comment by arkeo01** [comment]
I would like to take up this issue, may I?

**Comment by lubber-de** [comment]
I assigned it to you. Happy Coding and looking forward to a PR 🙂

**Comment by snehadr23** [comment]
Assuming there is another calendar input for just date *before* the user selects time that selected date, the below code can disable the every other hour except for 10AM - 5PM using *minDate* and *maxDate* when these lines are in '**onChange**' callback for the date input calendar

```
$('.ui.calendar#time_calendar')
        .calendar('set minDate', 
            new Date(selectedDate.getFullYear(), 
                        selectedDate.getMonth(), 
                        selectedDate.getDate(), 
                        10)
                );
    $('.ui.calendar#time_calendar')
        .calendar('set maxDate', 
            new Date(selectedDate.getFullYear(),
                     selectedDate.getMonth(),
                     selectedDate.getDate(),
                     17)
            );
```

Note:

1. **selectedDate** in the lines above is the date selected by the user before selecting time.
2. To ensure the user selects a date first, time second - I chose to disable the time calendar input until a date is selected.

**Comment by lubber-de** [comment]
Implemented by #2005
